connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -407,3 +409,18 @@ message NamedArgumentExpression {
   // (Required) The value expression of the named argument.
   Expression value = 2;
 }
+
+message Origin {
```

Comment:
this needs to be moved to a common place as it's not specific to Expressions.

Reply:
The current implementation of tracking lazy evaluation is specific to `Expression`s actually (see `org.apache.spark.sql.withOrigin`.

Reply:
Re: https://github.com/apache/spark/pull/46789#discussion_r1640210979 (I think it's related).

The tricky thing is that `RelationCommon` binds to `Relation`. However, the lazy evaluation tracking is based on where the expressions were created ,e.g., before we create a `DataFrame`.

If we can store the callside of the individual expressions at `source_info` in `RelationCommon`, we could still be able to do it, but I think it's a little bit awkward probably because the current implementation is specific to `Expression`s.

Reply:
My point is different. The callsite origin is something that is very generic and very useful for relations as well. That the origin today is only applicable to expressions is a side-effect of the implementation.

My goal is to have something like this

```
message Origin {
}

message Relation {
   RelationCommon common = 1;
   message RelationCommon {
      Origin origin = 1;
   }
}

message Expression {
   ExpressionCommon common = 1;
   message ExpressionCommon {
      Origin origin = 1;
   }
}

```

Reply:
Ah sure.